src/m365/planner/commands/task/task-checklistitem-remove.ts

```diff
@@ -0,0 +1,105 @@
+import { Cli, Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id: string;
+  taskId: string;
+  confirm?: boolean;
+}
+
+class PlannerTaskChecklistItemRemoveCommand extends GraphCommand {
+  public get name(): string {
+    return commands.TASK_CHECKLISTITEM_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes the checklist item from the planner task';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.confirm = (!(!args.options.confirm)).toString();
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (args.options.confirm) {
+      this.removeChecklistitem(logger, args, cb);
+    }
+    else {
+      Cli.prompt({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to remove the checklist item with id ${args.options.id} from the planner task?`
+      }, (result: { continue: boolean }): void => {
+        if (!result.continue) {
+          cb();
+        }
+        else {
+          this.removeChecklistitem(logger, args, cb);
+        }
+      });
+    }
+  }
+
+  private removeChecklistitem(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    this
+      .getTaskDetailsEtag(args.options.taskId)
+      .then(etag => {
+        const requestOptionsTaskDetails: any = {
```

Comment:
@martinlingstuyl - Thank you for reviewing the changes.
The graph API does not throw any error if the checklist item is not present. It actually deletes the item if present otherwise it simply ignores it. Let me know if we still need to check if it really exists. 
Thank you!

Reply:
I'm not talking about the graph. What I mean is that you should check client side _if the task contains the specified checklistitem_. If not, the code should throw an error saying the checklistitem cannot be found.

Reply:
Got you. I will implement the change and let you know once done.

Reply:
@martinlingstuyl - Added a validation to check if the id exists. Can you please review?

![image](https://user-images.githubusercontent.com/14875265/174302229-2124f73a-c240-44a8-a90f-93467f295dad.png)